browser/lib/transport/xhrtransport.js

```diff
@@ -9,7 +9,7 @@ var XHRTransport = (function() {
 	XHRTransport.isAvailable = XHRRequest.isAvailable;
 
 	XHRTransport.checkConnectivity = function(callback) {
-		Http.Request('http://internet-up.ably.io.s3-website-us-east-1.amazonaws.com/is-the-internet-up.txt', null, null, null, function(err, responseText) {
+		Http.Request(Defaults.internetUpUrlWithoutExtension + '.txt', null, null, null, function(err, responseText) {
```

Comment:
> can you confirm if this check is only made when in a browser?

Also in the node comet transport.

> If not, I will need to append the current protocol to this request when in a browser environment I assume.

I don't know what this means.

> Also, is internet connectivity only checked when falling back to XHR & JSONP? I assume that is intentional.

Since we don't fall back but we fall forward, upgrade to web socket never occurs until one of the HTTP transports connects. So we only use the HTTP transports to check connectivity for the purposes of deciding whether or not to use a fallback host.